### DISCUSSION TITLE
Problems with phone type audios

### DISCUSSION THREAD

**Discussion by evilc3** [NVIDIA-NeMo/NeMo#4483] [Q&A]
Hi, I hv trained a new conformer model. 

For telephone audios this model performs worse than the pretrained model used for fine-tuning it. 

For Other audios its ok 

Error seen are quite odd instead of insertion or substitution all the errors deletion. It just doesn't transcribe parts of the audio that pretrain model does. 

1. Changes made for training my model 
     - have changed tokens from 128 to 512 added. Puncts. (Same tok type used unigram). 
     
     - Trained on different dataset of size ~10k  hrs. (Amond them fisher was the only telephone speech dataset)

     
     
I am unable to debug where/why this is happening  

Possible ans 
1. Turning data hd problems. 
2. Not enough telephone type training data used. Eg ( asr is not able to transcribe becz the audio is noisy )

Can anyone please pont me in the write direction.

**Comment by okuchaiev** [comment]
I wonder why you changed tokens? I would try not changing the vocabulary.
Also, please share more details on LR, training plots etc.

**Reply by evilc3** [reply]
Tokens were changed to include puncts. And this hs worked well. It includes puncts without the. Use of additional punct model.  

Lr = 2, batch_size =128

Training loss and val wer were correctly decreasing.

I think this problem is with the audio type, it works as expected with clean audios.

**Reply by titu1994** [reply]
Isn't fisher around 200 or so hours? Probably model just didn't learn much or wasn't trained enough on 10k hours of speech. We generally train for ~ 100-200k steps depending on the size of the model (smaller models take more steps). Even then it's not guaranteed that it will always be noise robust, you can try noise augmented training but it will hurt clean audio wer. 

Finally, the noise level of your eval data may just be something the model has never seen - ie Fisher is insufficient to train it for such scenarios

**Reply by evilc3** [reply]
@titu1994 thanks for the reply.So ur saying fisher data alone is not enough  right

An update, so I was able to get  little better results when I increased my beam width it was set to low. (Prev beam width was 5, new 100). It has reduced the deletion problem quite a bit. 

Default decoding I used was kenlm model. Pyctcdecode version, alpha = 0.5, beta = 10, 
Beam_width = 5

A thing to note with just greedy decoding the outputs were better then with beam_width 5.

But pretrained model is able to get better results even with setting beam_width set to 5. 

Also tried a experiment, augmenting an audio by adding noise to it. The deletion problem occured there to. So yes looks like will need to make model more robust for noisy data. 

Here is my strategy 

Using adapters train a model with noisy augmented data. With adapters the forgetting problem can be eliminated . 

And if audio is phone type I can just set the adapter to true else false. 


What do ul think abt this?

I hv looked at a couple of augs, I want a aug that mimics phone calls any suggestions.

**Reply by titu1994** [reply]
That sounds reasonable, note that forgetting problem is not perfectly solved. If you overfit the audio domain with the adapter, when you switch on the adapter your wer on clean speech may still increase. 

For noise augmentation, you should use the G.37 augmentation or simply add background noise with some probability. @jbalam-nv for more info on specific audio augmentation

**Reply by evilc3** [reply]
What's G.37?

**Reply by titu1994** [reply]
Woops not G.37, it's G711 https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/parts/preprocessing/perturb.py#L628
